instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java

```diff
@@ -61,8 +61,14 @@ public void onEnd(
     String peerService = mapToPeerService(peerName);
     if (peerService != null) {
       attributes.put(SemanticAttributes.PEER_SERVICE, peerService);
+    } else {
+      // try map sockPeerName only if there was no match for peerName
+      String sockPeerName = attributesGetter.sockPeerName(request, response);
+      String sockPeerService = mapToPeerService(sockPeerName);
+      if (sockPeerService != null) {
+        attributes.put(SemanticAttributes.PEER_SERVICE, sockPeerService);
+      }
     }
```

Comment:
Nit: you could restructure this code a little a bit and have just one `attributes.put` call:
```suggestion
    if (peerService == null) {
      String sockPeerName = attributesGetter.sockPeerName(request, response);
      peerService = mapToPeerService(sockPeerName);
    }
    if (peerService != null) {
      attributes.put(SemanticAttributes.PEER_SERVICE, peerService);
    }
```